core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -320,34 +419,35 @@ LookUp lookUp(K key) {
             }
             // if keys are equal - we've found the item
             else if (cmp == 0) {
-                int hi = getEntryField(curr, OFFSET_HANDLE_INDEX);
-                if (hi < 0) {
-                    return new LookUp(null, curr, -1);
+                long valueReference = getValueReference(curr);
+                Slice valueSlice = buildValueSlice(valueReference);
+                if (valueSlice == null) {
+                    assert valueReference == 0;
+                    return new LookUp(null, valueReference, curr);
                 }
-                Handle h = handles[hi];
-                if (h.isDeleted()) {
-                    return new LookUp(null, curr, hi);
+                if (ValueUtils.isValueDeleted(valueSlice)) {
```

Comment:
Please add a comment that even if slice is not null the value can be still marked as deleted, what is checked here...